airflow/operators/python.py

```diff
@@ -190,7 +193,15 @@ def _get_unique_task_id(task_id: str, dag: Optional[DAG] = None) -> str:
           task_id__20
         """
         dag = dag or DagContext.get_current_dag()
-        if not dag or task_id not in dag.task_ids:
+        if not dag:
+            return task_id
+
+        # We need check if we are in context of TaskGroup as the task_id may
+        # already be altered
+        task_group = task_group or TaskGroupContext.get_current_task_group(dag)
+        tg_task_id = task_group.child_id(task_id) if task_group else None
+
+        if (task_id not in dag.task_ids) and (tg_task_id not in dag.task_ids):
```

Comment:
Oh damn, that makes sense!

Reply:
I think this is equivalent and simpler (because tasks not in a user given `TaskGroup` are in the root `TaskGroup` and have the original `task_id`):
```python
        # We need to check if we are in the context of TaskGroup as the task_id may
        # already be altered
        task_group = task_group or TaskGroupContext.get_current_task_group(dag)
        tg_task_id = task_group.child_id(task_id) if task_group else task_id

        if tg_task_id not in dag.task_ids:
            return task_id
        ...
``````

Reply:
Yup, that make sense!